lib/screens/v2/candidate_generator/dup/departures.ex

```diff
@@ -59,70 +62,127 @@ defmodule Screens.V2.CandidateGenerator.Dup.Departures do
+        &get_departure_instance_for_section(
+          &1,
+          length(sections_data) == 1,
+          now,
+          fetch_schedules_fn
+        )
       )
 
     Enum.map(slot_ids, fn slot_id ->
-      if Enum.all?(sections, &(&1.type == :no_data_section)) do
-        %DeparturesNoData{
-          screen: config,
-          slot_name: slot_id
-        }
-      else
-        %DeparturesWidget{
-          screen: config,
-          section_data: sections,
-          slot_names: [slot_id]
-        }
+      cond do
+        Enum.all?(sections, &(&1.type == :no_data_section)) ->
+          %DeparturesNoData{
+            screen: config,
+            slot_name: slot_id
+          }
+
+        Enum.all?(sections, &(&1.type == :overnight_section)) ->
+          %OvernightDepartures{
+            screen: config,
+            slot_names: [slot_id],
+            routes: get_route_pills_for_rotation(sections)
+          }
+
+        true ->
+          %DeparturesWidget{
+            screen: config,
+            section_data: sections,
+            slot_names: [slot_id]
+          }
       end
     end)
   end
 
-  defp get_departure_instance_for_section(%{type: :no_data_section} = section, _, _), do: section
+  defp get_departure_instance_for_section(%{type: :no_data_section} = section, _, _, _),
+    do: section
 
   defp get_departure_instance_for_section(
          %{
            departures: departures,
            alert: alert,
            headway: headway,
            stop_ids: stop_ids,
-           route: route
+           routes: routes,
+           params: params
          },
          is_only_section,
-         now
+         now,
+         fetch_schedules_fn
        ) do
-    visible_departures =
-      if is_only_section do
-        Enum.take(departures, 4)
-      else
-        Enum.take(departures, 2)
-      end
+    stops_with_live_departures = departures |> Enum.map(&Departure.stop_id/1) |> Enum.uniq()
+
+    overnight_schedules_for_section =
+      get_overnight_schedules_for_section(
+        stops_with_live_departures,
+        params,
+        routes,
+        alert,
+        now,
+        fetch_schedules_fn
+      )
+
+    headway_mode = get_headway_mode(stop_ids, headway, alert, now)
+
+    cond do
+      # All routes in section are overnight
+      overnight_schedules_for_section != [] and departures == [] ->
+        %{type: :overnight_section, routes: routes}
+
+      # There are still predictions to show
+      headway_mode == :inactive ->
+        # Add overnight departures to the end.
+        # This allows overnight departures to appear as we start to run out of predictions to show.
+        departures = departures ++ overnight_schedules_for_section
+
+        visible_departures =
+          if is_only_section do
+            Enum.take(departures, 4)
+          else
+            Enum.take(departures, 2)
+          end
+
+        if visible_departures == [] do
+          %{type: :no_data_section, route: List.first(routes)}
```

Comment:
This is something that PaulH is still unsure about and will sort out in QA. This was an assumption on my part because if there are no `visible_departures` and no alerts, the section would just be blank. There may be another state that design will come up with, but for now this is enough. 

As for getting the first route, I don't believe we mix and match route types in any section. So grabbing the first route should be enough to convert that to a pill (with CR and Bus showing an icon pill).